arch/Kconfig

```diff
@@ -425,6 +425,12 @@ config CPU_HAS_TEE
 	  Execution Environment (e.g. when it has a security attribution
 	  unit).
 
+config CPU_HAS_DCLS
```

Comment:
Feels like a Cortex-R specific config option, should this be in `arch/arm/core/cortex_r/Kconfig `?

Reply:
At the time of writing, Dual-redundant Core Lock-step (DCLS) is also available in the following cores that are not Cortex-R:
* Cortex-M23
* Cortex-M33
* Cortex-M35P
* Cortex-M7
* Cortex-A65AE
* Cortex-A76AE

Reply:
Had no idea, so maybe then it should be in the `arch/arm/core/Kconfig`

Reply:
I was thinking `CPU_HAS_DCLS` could be a global symbol that describes "Dual-redundant Core Lock-step" feature for all architectures, like `CPU_HAS_TEE` and `CPU_HAS_MPU`.

After all, Dual-redundant Core Lock-step is not an ARM-specific feature and there are many other architectures that implement this. So instead of having different symbols for representing the same feature across many architectures, it would be better to have one symbol that describes it.